credentials/java/com/google/auth/Credentials.java

```diff
@@ -36,6 +37,37 @@
   }
 
   /**
+   * Get the current request metadata without blocking.
+   *
+   * <p>This should be called by the transport layer on each request, and the data should be
+   * populated in headers or other context. The implementation can either call the callback inline,
+   * or from the given executor asynchronously. Either way it should <strong>never block</strong> in
```

Comment:
How about removing "from the given executor"? Just "or asynchronously. There shouldn't be a requirement that the executor is used when doing async.